pkgs/development/python-modules/knocki/default.nix

```
@@ -0,0 +1,62 @@
+{
+  aiohttp,
+  aioresponses,
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  mashumaro,
+  orjson,
+  poetry-core,
+  pythonOlder,
+  pytestCheckHook,
+  pytest-aiohttp,
+  syrupy,
+  yarl,
+}:
+
+buildPythonPackage rec {
+  pname = "knocki";
+  version = "0.3.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.11";
+
+  src = fetchFromGitHub {
+    owner = "swan-solutions";
+    repo = "knocki-homeassistant";
+    rev = "v${version}";
+    hash = "sha256-tWtANsujAcdIddoUBrKWIPfiPzDqWhW94Goz0QQ2BfE=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "addopts = \"--cov\"" ""
+  '';
+
+  nativeCheckInputs = [
+    aioresponses
+    pytestCheckHook
+    pytest-aiohttp
+    syrupy
+  ];
+
+  dependencies = [
+    aiohttp
+    mashumaro
+    orjson
+    yarl
+  ];
```

```suggestion
  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace-fail "addopts = \"--cov\"" ""
  '';

  build-system = [ poetry-core ];

  dependencies = [
    aiohttp
    mashumaro
    orjson
    yarl
  ];

  nativeCheckInputs = [
    aioresponses
    pytestCheckHook
    pytest-aiohttp
    syrupy
  ];
```

Bringing them into the order as they are used, which is

- patch
- build
- test